tests/bgp/test_bgp_session_flap.py

```diff
@@ -0,0 +1,204 @@
+            float(parsed_ipv6[0]), float(parsed_ipv6[1]), float(parsed_ipv6[2])]
+    return data
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, enum_asic_index):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    tor_neighbors = dict()
+    tor1 = natsorted([neighbor for neighbor in nbrhosts.keys() if neighbor.endswith('T0')])[0]
+
+    bgp_facts = duthost.bgp_facts(instance_id=enum_asic_index)['ansible_facts']
+    neigh_asn = dict()
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'] not in skip_hosts:
+            neigh_asn[v['description']] = v['remote AS']
+            tor_neighbors[v['description']] = nbrhosts[v['description']]["host"]
+            assert v['state'] == 'established'
+
+    # verify sessions are established
+    logger.info(duthost.shell('show ip bgp summary'))
+    logger.info(duthost.shell('show ipv6 bgp summary'))
+
+    setup_info = {
+        'duthost': duthost,
+        'neighhost': tor_neighbors[tor1],
+        'neigh_asn': neigh_asn[tor1],
+        'asn_dict':  neigh_asn,
+        'neighbors': tor_neighbors
+    }
+
+    logger.info("DUT BGP Config: {}".format(duthost.shell("show run bgp", module_ignore_errors=True)))
+    logger.info("Neighbor BGP Config: {}".format(
+        nbrhosts[tor1]["host"].eos_command(commands=["show run | section bgp"])))
+    logger.info('Setup_info: {}'.format(setup_info))
+
+    # get baseline BGP CPU and Memory Utilization
+    get_cpu_stats(duthost)
+
+    yield setup_info
+
+    global stop_threads
+    global flap_threads
+    stop_threads = True
+    time.sleep(wait_time)
+
+    # ensure no threads are left running
+    for thread in flap_threads:
+        thread_exception = thread.join(timeout=0.1,
+                                       suppress_exception=True)
+        if thread_exception:
+            logger.debug("Exception in thread %r:", thread)
+            logger.debug(
+                "".join(traceback.format_exception(*thread_exception))
+            )
+    flap_threads = []
+
+
+def flap_neighbor_session(neigh, asn):
+    while(True):
+        neigh.kill_bgpd()
+        neigh.start_bgpd()
+        if stop_threads:
+            logger.info("stop_threads now true, breaking loop")
+            break
+
+
+def test_bgp_single_session_flaps(setup):
+    global flap_threads
+    global stop_threads
+    stop_threads = False
+
+    # get baseline stat information
+    stats = []
+    stats.append(get_cpu_stats(setup['duthost']))
+
+    # start threads to flap neighbor sessions
+    thread = InterruptableThread(
+        target=flap_neighbor_session,
+        args=(setup['neighhost'], setup['neigh_asn']))
+    thread.daemon = True
+    thread.start()
+    flap_threads.append(thread)
+
+    for i in range(10):
+        stats.append(get_cpu_stats(setup['duthost']))
+        index = len(stats) - 1
+        assert stats[index][0] < (stats[index - 1][0] + 10)
```

Comment:
Could you explain whats the thought process behind this calculation? line no. 134-142. In a single session flap, why would memory/cpu load keep incresing from prev iteration?

Reply:
This is to ensure that memory or CPU doesn't spike during the flap.

Reply:
Can you explain further? >> the thought process behind this calculation? line no. 134-142. In a single session flap, why would memory/cpu load keep increasing from prev iteration?

Reply:
I updated to compare to the baseline as per our conversation.  I also moved the 10 and 1.3 to variables at the top for easier customization.